controllers/imagerepository_controller.go

```diff
@@ -358,6 +359,22 @@ func (r *ImageRepositoryReconciler) ProvisionImageRepository(ctx context.Context
 			// Do not brake provision because of failed owner reference
 		}
 	}
+
+	if isComponentLinked(imageRepository) {
+		updateComponentAnnotation, updateComponentAnnotationExists := imageRepository.Annotations[updateComponentAnnotationName]
+		if updateComponentAnnotationExists && updateComponentAnnotation == "true" {
+			component.Spec.ContainerImage = quayImageURL
+
+			if err := r.Client.Update(ctx, component); err != nil {
+				log.Error(err, "failed to update Component after provision")
+				return err
+			} else {
+				log.Info("Updated component's ContainerImage")
+				delete(imageRepository.Annotations, updateComponentAnnotationName)
+			}
+		}
+	}
```

Comment:
I'd place it in the `Reconcile` (right before metrics), because `Component` could be changed, so we'll have to redo all the provision checks again without real reason for it.

Reply:
I've added updating component actually after provision, because at that time image repository will be provisioned already and another reconcile will trigger and there we will take care of updating component and also remove that component update annotation

Reply:
I don't think I'm able to get your point from these words. But, what I figure out is the successful image repository provision adds the `image-repository` finalizer, once Component update fails, the Component update will not run in another Reconcile triggered by the failure because of the existing finalizer.